exp/runtime/api/v1alpha1/conversion.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apimachineryconversion "k8s.io/apimachinery/pkg/conversion"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta2"
+	runtimev1 "sigs.k8s.io/cluster-api/exp/runtime/api/v1beta2"
+)
+
+func (src *ExtensionConfig) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*runtimev1.ExtensionConfig)
+
+	return Convert_v1alpha1_ExtensionConfig_To_v1beta2_ExtensionConfig(src, dst, nil)
+}
+
+func (dst *ExtensionConfig) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*runtimev1.ExtensionConfig)
+
+	return Convert_v1beta2_ExtensionConfig_To_v1alpha1_ExtensionConfig(src, dst, nil)
+}
+
+func Convert_v1beta2_ExtensionConfigStatus_To_v1alpha1_ExtensionConfigStatus(in *runtimev1.ExtensionConfigStatus, out *ExtensionConfigStatus, s apimachineryconversion.Scope) error {
+	if err := autoConvert_v1beta2_ExtensionConfigStatus_To_v1alpha1_ExtensionConfigStatus(in, out, s); err != nil {
+		return err
+	}
+
+	// Reset conditions from autogenerated conversions
+	// NOTE: v1beta2 conditions should not be automatically be converted into legacy conditions (v1beta1).
+	out.Conditions = nil
+
+	// Retrieve legacy conditions (v1beta1) from the deprecated field.
+	if in.Deprecated != nil && in.Deprecated.V1Beta1 != nil {
+		if in.Deprecated.V1Beta1.Conditions != nil {
+			out.Conditions = in.Deprecated.V1Beta1.Conditions
```

Comment:
This should not be possible without conversion. v1alpha1 should not be using v1beta2.Conditions. Looks like this was changed on main (on release-1.10 v1alpha1 is using v1beta1.Conditions)

(e.g. CRS is using `clusterv1beta1.Convert_v1beta2_Deprecated_V1Beta1_Conditions_To_v1beta1_Conditions(&in.Deprecated.V1Beta1.Conditions, &out.Conditions)`)